### ISSUE TITLE
Facebook: Sending typing indicator

### ISSUE THREAD

**Issue by mhjb** [howdyai/botkit#1728]
### Context:

- Botkit version: 4.0.2
- Messaging Platform: Messenger
- Node version: 12.4.0
- Os: MacOS 10.14.5

On Botkit v0, the following worked:

`bot.reply(message, { sender_action: 'typing_on' })`

On Botkit v4, that produces the error:

`Error sending activity to Facebook: (#100) Cannot send both message and state at the same time`

With the web adaptor, I can use 

`bot.reply(message, { type: 'typing' })`

… but that does nothing with the Facebook adaptor in v4.

Any clues?

**Comment by yestouscool** [comment]
Yes, I have the same problem and all facebook custom template : quick replies and othe json format message are not being sent.

**Comment by LuckyLuo** [comment]
@mhjb  Is tpying indicator working for you now?
@benbrown I upgraded botkit to 4.5, but typing indicator still doesn't work.
`controller.on('message,direct_message', async (bot, message) => {
  await bot.reply(message, { sender_action: 'typing_on' });
  await bot.reply(message, 'test for typing indicator');
}
`
Do I make any mistake here? Thanks

**Comment by mhjb** [comment]
hi @benbrown sorry for slow response — I've upgraded to botkit 4.5. Neither

`bot.reply(message, { sender_action: 'typing_on' })`

nor

`bot.reply(message, { type: 'typing' })`

Work. Same errors as above. Looking at https://github.com/howdyai/botkit/blob/master/packages/botbuilder-adapter-facebook/src/facebook_adapter.ts#L271 it appears that the Facebook adaptor is not going to let through any activities with type 'typing'. 

Can you help?

**Comment by mhjb** [comment]
Figured it out… 

```
bot.api.callAPI('/me/messages', 'post', {
  recipient: {
    id: incoming_facebook_message.user
  },
  sender_action: 'typing_on'
})
```

**Comment by benbrown** [comment]
What do you know, I finally figured this out.

Turns out Facebook was rejecting the message because it contained extra (empty) fields.  

Fixed in v1.0.7

**Comment by benbrown** [comment]
For the record, you can now do:

`await bot.reply(message,{sender_action:'typing_on'});`

To use typing indicators in a dialog, use:

`convo.say({channelData: {sender_action: 'typing_on'}});`
